integration-tests/test_motd.py

```diff
@@ -0,0 +1,52 @@
+import contextlib
+import os
+
+import pytest
+
+
+DOT_REGISTERED_PATH = "/etc/insights-client/.registered"
+DOT_UNREGISTERED_PATH = "/etc/insights-client/.unregistered"
+
+MOTD_PATH = "/etc/motd.d/insights-client"
+MOTD_SRC = "/etc/insights-client/insights-client.motd"
+
+
+pytestmark = pytest.mark.usefixtures("register_subman")
+
+
+@pytest.fixture
+def clean_system():
+    """Clean up the system before every test."""
+    with contextlib.suppress(OSError):
+        os.remove(DOT_REGISTERED_PATH)
+    with contextlib.suppress(OSError):
+        os.remove(DOT_UNREGISTERED_PATH)
+    with contextlib.suppress(OSError):
+        os.remove(MOTD_PATH)
+
+
+def test_motd(clean_system, insights_client):
+    """MOTD only exists on unregistered system without (.un)registered files."""
+    # If the system is not registered, the file should be present.
+    insights_client.run("--status")
+    assert os.path.exists(MOTD_PATH)
+
+    # After registration, the file should not exist.
+    insights_client.register()
+    assert not os.path.exists(MOTD_PATH)
+
+    # The system was unregistered. Because .unregistered file exists,
+    # the file should not be present.
+    insights_client.unregister()
+    assert not os.path.exists(MOTD_PATH)
+
+
+def test_motd_dev_null(clean_system, insights_client):
+    """MOTD should not be touched if it is a /dev/null symlink."""
+    os.symlink(os.devnull, MOTD_PATH)
```

Comment:
this symlink ought to be removed at the end; I'd recommend creating a `contextlib.ExitStack` variable, and pushing an helper function to `os.unlink()` the file